src/interface.c

```diff
@@ -22266,6 +22325,14 @@ int ascii_digest (hashcat_ctx_t *hashcat_ctx, char *out_buf, const size_t out_le
         ptr[15], ptr[14]
       );
   }
+  else if (hash_mode == 18100)
+  {
+      // salt_buf[1] holds our 32 bit value. salt_buf[0] and salt_buf[1] would be 64 bits.
+      // for now, we only need to worry about 32 bit counters.
+      // we also need to multiply salt by our step to see the floor of our original timestamp range.
+      // again, we will use the default RFC 6238 step of 30.
+      snprintf (out_buf, out_len - 1, "%06d:%d", digest_buf[1], byte_swap_32 (salt.salt_buf[1]) * 30);
```

Comment:
I found that the hashes can't be cracked multiple times this way (e.g. the output of hashcat can't be cracked again for certain hashes, e.g. "hashcat" -> 504267:85457442135216). My suggestion here is something like this:
```
      u64 tmp = (((u64) byte_swap_32 (salt.salt_buf[0])) << 32) | ((u64) byte_swap_32 (salt.salt_buf[1]));

      tmp *= 30;

      snprintf (out_buf, out_len - 1, "%06d:%lu", digest_buf[1], tmp);
```
(and remove the comment about the 32 bit is enough)
Note: we need 64 bit because the * 30 could also overflow etc